src/pyproject_fmt/__main__.py

```diff
@@ -25,14 +25,14 @@ def color_diff(diff: Iterable[str]) -> Iterable[str]:
 
 def run(args: Sequence[str] | None = None) -> int:
     opts = cli_args(sys.argv[1:] if args is None else args)
-    formatted = format_pyproject(opts)
     toml = opts.pyproject_toml
-    before = toml.read_text()
+    before = toml.read_text(encoding="utf-8")
+    formatted = format_pyproject(before, opts)
     changed = before != formatted
     if opts.stdout:  # stdout just prints new format to stdout
         print(formatted, end="")
     else:
-        toml.write_text(formatted)
+        toml.write_text(formatted, encoding="utf-8")
```

Comment:
I explicitly added encoding (to both read and write) because the [TOML spec requires UTF-8](https://toml.io/en/v1.0.0#spec), but according to the [Python docs](https://docs.python.org/3/library/functions.html#open) the default encoding is [platform dependent](https://docs.python.org/3/library/locale.html#locale.getpreferredencoding)